lib/sockopt.c

```diff
@@ -23,8 +23,8 @@ void setsockopt_so_recvbuf(int sock, int size)
 {
 	int orig_req = size;
 
-	while (setsockopt(sock, SOL_SOCKET, SO_RCVBUF, &size, sizeof(size))
-	       == -1)
+	while (size && setsockopt(sock, SOL_SOCKET, SO_RCVBUF, &size,
+				  sizeof(size)) == -1)
```

Comment:
Do we know what the errno is?

I guess it's failing all the way down to 0, and then your fix is keeping things from forever looping?